server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -602,6 +613,15 @@ public void setMaxConcurrentShardRequests(int maxConcurrentShardRequests) {
         }
         this.maxConcurrentShardRequests = maxConcurrentShardRequests;
     }
+
+    public Map<String, long[]> getWaitForCheckpoints() {
+        return waitForCheckpoints;
+    }
+
+    public void setWaitForCheckpoints(Map<String, long[]> afterCheckpointsRefreshed) {
+        this.waitForCheckpoints = afterCheckpointsRefreshed;
```

Comment:
The validation (`validate`) should be updated to prevent this option to be used with a PIT (point in time reader). 
Unrelated but I also wonder if the feature could be implemented using a PIT. We could allow the wait_for_checkpoint option when opening a PIT and then the caller have the guarantee that the PIT is valid for every search request ? We also don't need a new search API if we take that path.

Reply:
Done. I talked to Henning and the PIT is interesting. But might make more sense if we want to continue to improve this in the future for more use cases. It would be a pretty dramatic departure pretty late in the cycle for what we had planned here.

Reply:
sure, just a late idea I'd say. Let's continue with the current approach.